workflow/artifacts/http/http_test.go

```diff
@@ -64,11 +62,10 @@ func TestArtifactoryArtifactDriver_Load(t *testing.T) {
 				Artifactory: &wfv1.ArtifactoryArtifact{URL: "https://github.com/argoproj/argo-workflows/not-found"},
 			},
 		}, "/tmp/not-found")
-		if assert.Error(t, err) {
-			argoError, ok := err.(errors.ArgoError)
-			if assert.True(t, ok) {
-				assert.Equal(t, errors.CodeNotFound, argoError.Code())
-			}
+		require.Error(t, err)
+		argoError, ok := err.(errors.ArgoError)
+		if assert.True(t, ok) {
```

Comment:
same as above.

I also wonder if this is further simplifiable with the `Error*` helpers